### PR TITLE
FINERACT-1411 Allow charges type saving to map to all types of Ledgers

### PR DESCRIPTION
**Pull request by ankita10r**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/accounting/common/AccountingDropdownReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
     private final GLAccountReadPlatformService accountReadPlatformService;
 
     @Autowired
-    public AccountingDropdownReadPlatformServiceImpl(final GLAccountReadPlatformService accountReadPlatformService) {
+    public AccountingDropdownReadPlatformServiceImpl(GLAccountReadPlatformService accountReadPlatformService) {
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/accounting/common/AccountingDropdownReadPlatformServiceImpl.java]: Assign final.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountChargeReadPlatformServiceImpl.java [hunk 2]
```diff
@@ -185,6 +191,23 @@ public SavingsAccountChargeData retrieveSavingsAccountChargeDetails(final Long i
         }
     }
 
+    @Override
+    public long retrieveSavingsAccountChargeDetailsId(final Long id, final Long savingsAccountId) {
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountChargeReadPlatformServiceImpl.java]: This function is not called anywhere. Please remove unused code when sending the PR.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

**Comment by avikganguly01** [comment]
@ptuomola  @awasum  : Travis isn't running again. Is this related to bottlenecks on Apache Infra side? If yes, can we move this from Apache infra to some infrastructure which I can provide you access to?

Ankita has tried triggering the build in different ways by pushing commits but no luck so far. Gradle test locally (without the fail fast flag) is failing with some of the flaky tests communicated earlier. We have no way of knowing whether build is going to be triggered and whether it will pass after merging. I have ran the charge specific integration tests separately one by one and they are passing though. I would like to merge it based on the time sensitivity of this rollout in production but  I will hold onto it till you get an opportunity to comment.

**Comment by ptuomola** [comment]
@avikganguly01 Travis build triggered - let's see if it is passes

And for local builds you need to always start with a new empty database. Otherwise they will not pass.

**Comment by ptuomola** [comment]
Looks like the build failed due to Docker rate limit. Let's retry...

**Comment by ptuomola** [comment]
Looks like the build passed. That's great.

**Review by avikganguly01 — APPROVED** [review]

**Pull request merged**